apps/scan/frontend/src/api.ts

```diff
@@ -233,7 +237,14 @@ export const setTestMode = {
     const queryClient = useQueryClient();
     return useMutation(apiClient.setTestMode, {
       async onSuccess() {
+        // If doesUsbDriveRequireCastVoteRecordSync was true, switching from test mode to official
+        // mode resets it back to false. To avoid a flicker of the warning prompting you to sync
+        // CVRs before you can switch from official mode to test mode, we invalidate this query
+        // first.
```

Comment:
I did in fact find that, if I placed this query invalidation after the other query invalidations, I was still able to repro. Zooming out, it would be nice if there was a way to batch query invalidations and make a group of them atomic

Reply:
Hmm, that is interesting. I think the hard part that you're identifying is that key information is split between two queries. 

A few solution ideas:
- Don't render the warning while `mutation.isLoading` is true.
- Consolidate the info we need to a single query (this could be a new query that just answers the question of whether or not to show this warning)

I think the way you've solved it here is totally good enough for now too, but interesting problem!

Reply:
I like the idea of a consolidated query for situations like this! Basically accomplishes the batching that I describe in my original comment. (Will still merge as is for NH, though, to avoid a riskier change)